packages/wrangler/src/pages/publish.tsx

```diff
@@ -312,6 +317,12 @@ export const Handler = async ({
 		_redirects = readFileSync(join(directory, "_redirects"), "utf-8");
 	} catch {}
 
+	try {
+		// Developers can specify a custom _routes.json file, for projects with Pages
+		// Functions or projects in Advanced Mode
+		_routes = readFileSync(join(directory, "_routes.json"), "utf-8");
```

Comment:
Maybe we could rename this to `_routesCustom` and the generated routes var to `_routesGenerated` or something? It's kind of hard to follow all the optionality with the same variable name everywhere